### PR TITLE
=clt #18343 Improve structure of DistPubSub docs

### PR DESCRIPTION
**Pull request by patriknw**
I have split up the docs in two separate sections for Publish and Send delivery modes. Additional sample code and some minor improvements of the contents.

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18343

**Comment by patriknw** [comment]
@johanandren could you please review this PR

File: akka-docs/rst/java/distributed-pub-sub.rst [hunk 1]
```diff
@@ -22,43 +22,60 @@ a few seconds. Changes are only performed in the own part of the registry and th
...
-efficiency the message is sent over the wire only once per node (that has a matching topic),
-and then delivered to all subscribers of the local topic representation. This is the
-true pub/sub mode. A typical usage of this mode is a chat room in an instant messaging
-application.
+For efficiency the message is sent over the wire only once per node (that has a matching topic),
+and then delivered to all subscribers of the local topic representation.
 
-**4. DistributedPubSubMediator.Publish with sendOneMessageToEachGroup**
+You register actors to the local mediator with ``DistributedPubSubMediator.Subscribe``. 
+Successful ``Subscribe`` and ``Unsubscribe`` is acknowledged with
```
Comment (johanandren) [inline akka-docs/rst/java/distributed-pub-sub.rst]: Should it say what does successful mean here? Has it managed to subscribe to the local mediator, or subscribed and got gossiped to other nodes etc.?

File: akka-docs/rst/java/distributed-pub-sub.rst [hunk 2]
```diff
@@ -67,42 +84,65 @@ If all the subscribed actors have the same group name, then this works just like
...
-A Small Example
----------------
+You register actors to the local mediator with ``DistributedPubSubMediator.Put``.
+The ``ActorRef`` in ``Put`` must belong to the same local actor system as the mediator.
+The path without address information is the key to which you send messages.
+On each node there can only be one actor for a given path, since the path is unique
+within one local actor system.
 
-A subscriber actor:
+You send messages by sending ``DistributedPubSubMediator.Send`` message to the
```
Comment (johanandren) [inline akka-docs/rst/java/distributed-pub-sub.rst]: Should it mention what happens if the same path has been registered on multiple nodes?

File: akka-docs/rst/java/distributed-pub-sub.rst (hunk 1 above)
Comment (patriknw) [inline akka-docs/rst/java/distributed-pub-sub.rst]: good point, it only means that subscription has been registered at the local mediator

**Comment by johanandren** [comment]
LGTM (Excepts for those two notes)

**Comment by patriknw** [comment]
thanks

**Comment by patriknw** [comment]
I fixed Johan's notes

**Pull request merged**